File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2022s/IZ08IO1/Strategy/src/Sportsmen.java
public class Sportsmen
{
   private String surname;
   private final SportsmenStrategyJump sJump;
   public Sportsmen(String surname,String strJump)
   {
      this.surname=surname;
      sJump=SportsmenStrategyJumpFactory.get(strJump);
   }
   public String getSurname() { return surname; }
   public String Jump() { return sJump.execute(); }
   public String Run() { return "Running"; }
   public String Swim() { return "Swimming"; }
}

File: semestr-8/Jezyk C#/alexander/Csharp/Archiwum/2022s/ID06IO2/Strategy/Command/Sportsmen.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Strategy
{
    public class Sportsmen
    {
        //public string _name;
        //public string name { get { return _name; } private set { _name = value; } }
        public string name { get; private set; }
        private SportsmenStrategyJump jump;
        public Sportsmen(string name,SportsmenStrategyJump jump)
        {
            this.name = name;
            this.jump = jump;
        }
        public string Run() { return "Runing"; }
        public string Swim() { return "Swimming"; }
        public string Jump() { return jump.Jump(); }
        public string execute(SportsmenCommand command)
        {
            return name + ": " + command.execute(this);
        }
    }
}
